src/lazy/expanded/macro_evaluator.rs

```diff
@@ -432,47 +512,33 @@ impl<
 ///   (:values 1)                 => 1
 ///   (:values 1 2 3)             => 1 2 3
 ///   (:values 1 2 (:values 3 4)) => 1 2 3 4
-pub struct ValuesExpansion<'data, D: LazyDecoder<'data>, M: MacroInvocation<'data, D>> {
+pub struct ValuesExpansion<'top, 'data, D: LazyDecoder<'data>> {
     // Which argument the macro is in the process of expanding
-    arguments: M::ArgumentsIterator,
+    arguments: MacroExprArgsIterator<'top, 'data, D>,
```

Comment:
🗺️ Rather than holding an iterator over the syntactic elements from the invocation, the newly added `'top` lifetime allows us to hold an iterator over the _resolved_ arguments of the invocation. This handles variable resolution for us, which simplifies the implementation of each macro kind a good bit.